lib/kubernetes/src/api.ts

```diff
@@ -39,24 +50,51 @@ export const createResource = async (resource: K8sResource): Promise<void> => {
 };
 
 export const deleteResource = async (resource: K8sResource): Promise<void> => {
+  log.debug(
+    `deleteResource for %s(ns: %s) [dryrun=${dryRunKubeApiWrites}]`,
+    resource.name,
+    resource.namespace
+  );
+  if (dryRunKubeApiWrites) {
+    return;
+  }
   try {
     await resource.delete();
   } catch (e) {
     const err = kubernetesError(e);
     if (err.statusCode !== 404) {
-      log.error(err.message);
+      log.info(
+        "api err during deleteResource for %s(ns: %s): %s",
+        resource.name,
+        resource.namespace,
+        err.message
+      );
+    } else {
+      log.debug("deleteResource(): ignored 404 error");
```

Comment:
want to show the main error message here of the ignored error? :) is sometimes super valuable.

Reply:
It's just a generic Kubernetes 404 error message "<type> <name> does not exist". In this case, it's not useful at all and it's already quite noisy because we'll get a lot of these 404s on teardown